### PR TITLE
Remove all visreg code

### PR DESCRIPTION
**Pull request by shaleen76**
JIRA: 

### LGTM's
- [x] Dev LGTM
- [ ] Zoom LGTM

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ notifications:
 addons:
   firefox: '56.0'
   chrome: stable
+  sauce_connect: true
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply .travis.yml]: Removed.

File: .travis.yml [hunk 2]
```diff
@@ -20,6 +21,6 @@ before_script:
 - sh -e /etc/init.d/xvfb start
 - jdk_switcher use oraclejdk8
 - yarn start > /dev/null 2>&1 &
-- while ! curl --silent http://localhost:4444/wd/hub > /dev/null 2>&1; do sleep 1; done
+- while ! curl --silent http://localhost:3000/ > /dev/null 2>&1; do sleep 1; done
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply .travis.yml]: Reverted change

File: docs/components/typography/test/caption.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+---
+title: Typography
+minver: 0.1.6
+---
+{% extends 'component.njk' %}
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply docs/components/typography/test/caption.html]: extended test.njk

File: docs/components/typography/test/caption.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+title: Typography
+minver: 0.1.6
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply docs/components/typography/test/caption.html]: removed

File: docs/components/typography/test/caption.html [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+---
+{% extends 'component.njk' %}
+{% block content %}
+  <section><!-- Caption -->
+    <h2 id="caption">Caption</h2>
+    <p>
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply docs/components/typography/test/caption.html]: removed

File: test/package.json [hunk 6]
```diff
@@ -6,27 +6,27 @@
...
   "scripts": {
     "build": "tsc",
-    "clean": "npm run clean:visreg && npm run clean:regression && rm -rf built",
-    "clean:visreg": "rm -rf artifacts/visregScreenshots",
-    "clean:regression": "rm -rf artifacts/regressionScreenshots",
-    "lint": "tslint -c tslint.json visreg/",
+    "clean": "npm run clean:regression && npm run clean:built",
+    "clean:built": "rm -rf built",
+    "clean:regression": "npm run clean:regression:screenshots && npm run clean:regression:autoGeneratedTs && npm run clean:built",
+    "clean:regression:screenshots": "rm -rf artifacts/regressionScreenshots",
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply test/package.json]: removed

File: test/package.json [hunk 7]
```diff
@@ -6,27 +6,27 @@
...
   },
   "scripts": {
     "build": "tsc",
-    "clean": "npm run clean:visreg && npm run clean:regression && rm -rf built",
-    "clean:visreg": "rm -rf artifacts/visregScreenshots",
-    "clean:regression": "rm -rf artifacts/regressionScreenshots",
-    "lint": "tslint -c tslint.json visreg/",
+    "clean": "npm run clean:regression && npm run clean:built",
+    "clean:built": "rm -rf built",
+    "clean:regression": "npm run clean:regression:screenshots && npm run clean:regression:autoGeneratedTs && npm run clean:built",
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply test/package.json]: removed

File: test/common/util.ts [hunk 8]
```diff
@@ -19,7 +19,7 @@ export async function sleep(ms = 1500) {
 export async function webComponentsReady(driver: WebDriver) {
     const ready = async () => (await driver.executeScript("return window.WebComponents.ready")) as boolean;
     while (!await ready()) {
-        await sleep(100);
+        await sleep(1000);
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply test/common/util.ts]: change reverted

**Comment by Droogans** [comment]
Don't forget to squash this down as to remove the saucelabs branch's commits.

Afterwards, please remove the edge, IE, and safari snapshots too.

**Comment by shaleen76** [comment]
Squashed all commits to 1 and removed IE, edge, safari snapshots

File: test/dom-snapshots/elements-hx-progress-chrome.js.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
+# Snapshot report for `built/dom-snapshots/elements-hx-progress-chrome.js`
+
+The actual snapshot is saved in `elements-hx-progress-chrome.js.snap`.
+
+Generated by [AVA](https://ava.li).
+
+## chrome on macOS 10.13 -- elements/hx-progress/test.html
```
Comment (Droogans) [inline test/dom-snapshots/elements-hx-progress-chrome.js.md]: If you're running these tests locally on saucelabs, you should stop. The travis-ci build server will run the browser locally using a local selenium server....which means the initial checked in version must do the same.

File: test/regression/runSingleTest.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import {test, TestContext} from "ava";
+import {By, snap, Snappit, IConfig, WebDriver, WebElement} from "snappit-visual-regression";
+
+import * as util from "../common/util";
+
+export async function regressionTest(browser: string, component: string) {
+    
+    const config: IConfig = {
+        browser: browser,
+        threshold: 0.1,
```
Comment (Droogans) [inline test/regression/runSingleTest.ts]: This can go. `threshold` is for images.

Comment (shaleen76) [reply test/regression/runSingleTest.ts]: removed

File: test/behavioral/visreg.ts [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+import {test} from "ava";
+
+export function suite(browserName: string) {
+    let snappit: Snappit;
+    let driver: WebDriver;
+
+    test.before(async () => {
+        const config: IConfig = {
+            browser: browserName,
+            threshold: 0.1,
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply test/behavioral/visreg.ts]: removed

File: docs/components/typography/test/inline-code.html [hunk 12]
```diff
@@ -0,0 +1,8 @@
+---
+title: Typography
+minver: 0.1.6
+---
+{% extends 'component.njk' %}
+{% block content %}
+  
+{% endblock %}
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply docs/components/typography/test/inline-code.html]: deleted

File: test/dom-snapshots/README.md [hunk 13]
```diff
@@ -0,0 +1,3 @@
+# dom-snapshots
+
+This is where test files for auto-generated regression tests end up. This then leaves behind dom snapshot artifacts next to them, which are *not* ignored by version control. **The test files, however, are ignored by version control**. They are simply generated by the parent test runner script at runtime. It uses standard boilerplate to import a top-level test that simply looks for elements in the dom that are flagged for regression testing, and checks to ensure that none of those elements have had their dom changed.
```
[the remark this answers is not in the document]
Comment (shaleen76) [reply test/dom-snapshots/README.md]: updated the Readme.md file

File: docs/components/buttons/index.html [hunk 14]
```diff
@@ -83,7 +83,7 @@ <h3>Button Group</h3>
     </div>
   </section>
 
-  <section data-visreg="button-states">
+  <section data-dom-diff="button-states">
```
Comment (CITguy) [inline docs/components/buttons/index.html]: Remove testing attribute. There is no Shadow DOM to test.

File: docs/components/typography/test/index.html [hunk 15]
```diff
@@ -0,0 +1,299 @@
...
+-          exit 2
+-        fi
+-
+-        genApacheConf(){
+-          echo -e "# Host ${HOME_DIR}$1/$2 :"
+-        }
+-      {% endcode %}
+    </figure>
+</section>
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 16]
```diff
@@ -0,0 +1,299 @@
+{% extends 'test.njk' %} {% block content %}
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 17]
```diff
@@ -0,0 +1,299 @@
...
+        <p class="hxCaption">Caption</p>
+    </div>
+</section>
+<section>
+    <!-- Error Message -->
+    <div class="demo" data-dom-diff="typography-error-message">
+        <hx-error>Error message goes here</hx-error>
+    </div>
+</section>
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 18]
```diff
@@ -0,0 +1,299 @@
+{% extends 'test.njk' %} {% block content %}
+<section>
+    <!-- Caption -->
+    <div class="demo" data-dom-diff="typography-caption">
+        <p class="hxCaption">Caption</p>
+    </div>
+</section>
+<section>
+    <!-- Error Message -->
+    <div class="demo" data-dom-diff="typography-error-message">
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Would it make more sense to set the value of `data-dom-diff` to `hx-error` to align with the name of the custom element we're testing?

File: docs/components/typography/test/index.html [hunk 19]
```diff
@@ -0,0 +1,299 @@
...
+<section>
+    <!-- Inline Code -->
+    <div class="demo" data-dom-diff="typography-inline-code">
+        <p>
+            Add an event listener with the
+            <code>EventTarget.addEventListener()</code> function.
+        </p>
+    </div>
+</section>
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 20]
```diff
@@ -0,0 +1,299 @@
...
+    <!-- Keyboard -->
+    <div class="demo" data-dom-diff="typography-keyboard">
+        <p>
+            Press
+            <kbd>Ctrl+Z</kbd> (
+            <kbd>Command+Z</kbd> on Mac) to undo the most recent action.
+        </p>
+    </div>
+</section>
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 21]
```diff
@@ -0,0 +1,299 @@
...
+            <li>
+                <a href="#">Normal Link</a>
+            </li>
+            <li>
+                <a href="#" class="pseudo-active">Pressed Link</a>
+            </li>
+        </ul>
+    </div>
+</section>
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 22]
```diff
@@ -0,0 +1,299 @@
...
+    <!-- Sub-Body -->
+    <div class="demo" data-dom-diff="typography-sub-body">
+        <p class="hxSubBody">Sub-Body</p>
+    </div>
+</section>
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 23]
```diff
@@ -0,0 +1,299 @@
...
+    <!-- Subdued -->
+    <div class="demo" data-dom-diff="typography-subdued">
+        <span class="hxSubdued">Subdued</span>
+    </div>
+</section>
+
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 24]
```diff
@@ -0,0 +1,299 @@
...
+-        input[type="checkbox"]:checked {
+-          /* ... */
+-        }
+-        .text-center {
+-          text-align: center !important;
+-        }
+-      {% endcode %}
+    </figure>
+</section>
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 25]
```diff
@@ -0,0 +1,299 @@
...
+        -
+        -        ! compress the size of the
+        -        ! changes.
+        -
+        -          It is important to spell
+        -      {% endcode %}
+    </figure>
+</section>
+
+<section><!-- Syntax Highlighting -->
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 26]
```diff
@@ -0,0 +1,299 @@
...
+          <p>
+            <img src="path/to/file.png" alt="pretty picture" />
+          </p>
+        </form>
+      {% endcode %}
+    </figure>
+    </section>
+
+
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 27]
```diff
@@ -0,0 +1,299 @@
...
+-
+-        [[lib]]
+-        name = ${NAME}
+-        default = True
+-        auto = no
+-        counter = 1_000
+-      {% endcode %}
+    </figure>
+</section>
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 28]
```diff
@@ -0,0 +1,299 @@
...
+-            return 'maybe';
+-          } else if (foo && bar) {
+-            return 'wut';
+-          }
+-          return `Hello ${name}!`;
+-        }
+-      {% endcode %}
+    </figure>
+</section>
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html [hunk 29]
```diff
@@ -0,0 +1,299 @@
...
+        -        > markdown is so cool
+        -
+        -            so are code segments
+        -
+        -        1. one thing (yeah!)
+        -        2. two thing `i can write code`, and `more` wipee!
+        -      {% endcode %}
+    </figure>
+</section>
+<section>
```
Comment (CITguy) [inline docs/components/typography/test/index.html]: Remove section. There is no Shadow DOM that needs testing.

File: docs/components/typography/test/index.html (hunk 16 above)
Comment (shaleen76) [reply docs/components/typography/test/index.html]: removed

File: docs/components/typography/test/index.html (hunk 25 above)
Comment (shaleen76) [reply docs/components/typography/test/index.html]: removed

File: docs/components/typography/test/index.html (hunk 26 above)
Comment (shaleen76) [reply docs/components/typography/test/index.html]: removed

File: docs/components/typography/test/index.html (hunk 27 above)
Comment (shaleen76) [reply docs/components/typography/test/index.html]: removed

File: docs/components/typography/test/index.html (hunk 28 above)
Comment (shaleen76) [reply docs/components/typography/test/index.html]: removed

File: docs/components/typography/test/index.html (hunk 29 above)
Comment (shaleen76) [reply docs/components/typography/test/index.html]: removed

File: docs/components/typography/test/index.html (hunk 19 above)
Comment (shaleen76) [reply docs/components/typography/test/index.html]: removed

File: docs/components/typography/test/index.html (hunk 20 above)
Comment (shaleen76) [reply docs/components/typography/test/index.html]: removed

File: docs/components/typography/test/index.html (hunk 21 above)
Comment (shaleen76) [reply docs/components/typography/test/index.html]: removed

File: docs/components/typography/test/index.html (hunk 24 above)
Comment (shaleen76) [reply docs/components/typography/test/index.html]: removed

**Comment by CITguy** [comment]
Why would the PUSH pass on Travis, but the PR is failing?

CC: @Droogans

**Comment by Droogans** [comment]
Squash this first please and rebase.

**Comment by CITguy** [comment]
Squashed. Waiting on Travis to approve.

**Pull request merged**